### ISSUE TITLE
Get/Set ID for a node

### ISSUE THREAD

**Issue by Zythyr** [svgdotjs/svg.js#839]
By default SVG.js assigns an ID to each node. How can I get the ID of that node? How can I change/set the ID of the node? I noticed there are functions to get/set the class of a node, but not an ID http://svgjs.com/manipulating/#classes. 

For example, I can manually inspect the SVG and find out that SVG.js has by default assigned the root SVG the ID #SvgjsSvg1001. What if I want to change the ID from #SvgjsSvg1001 to something like #MySVGCanvas?  How can I do this? 

`var draw = SVG('drawing')`

`<div id="drawing"><svg id="SvgjsSvg1001" width="100%" height="100%" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs"><defs id="SvgjsDefs1002"></defs></svg></div>`

Possible feature request: Is there a way we can customize the ID that SVG.js automatically assigns to all the nodes? For example I noticed that the ID assigned follows a specific pattern (ex: Svgjs[node-type][node-count]). Can we change this pattern?

**Comment by Fuzzyma** [comment]
You cannot change the pattern. However you can get or assign a new ID with the `id()` method.

We have very good docs at svgjs.com where a search for "ID" certainly would have revealed what you are searching for. Try the docs. They are awesome :)

**Comment by Zythyr** [comment]
@Fuzzyma  Yes I have been reading the docs svgjs.com. The id() method is not mentioned in the docs. Glad to know there is an id() method 👍  . This issue is resolved. 

Possible feature request for the future: Give us the ability to change the pattern to the automatically assigned ID. Possibly allowing us to add a prefix/suffix to the existing ID naming scheme?

**Comment by Fuzzyma** [comment]
Oh - that should not happen. We need to update the docs then. What you request as a feature most likely will be implemented in version 3.0

**Comment by Zythyr** [comment]
Thanks. Issue resolved.

**Comment by saivan** [comment]
I have had the same problem, where I needed to add prefixes and suffixes to the id's (I'm assuming because you have more than one svg with clashing id's in your document). To do this, I just made a function that does adds my chosen prefix directly.

**Comment by Fuzzyma** [comment]
Just for the records: id's created by svg.js will never clash as long as all id's are generated by svg.js

**Comment by saivan** [comment]
yes, I had this issue with an imported svg from some design software

**Comment by Zythyr** [comment]
@Fuzzyma Will thee be a clash (duplicated ID) issue if I clone a node onto another canvas and manually set the ID? 
See example below: 
```
nodeB = nodeA.clone().clear().addTo(parentNodeB)
nodeB.id(nodeA.id())
```
A is on SVG canvas A
B is on SVG canvas B
// Assuming is that canvas B does NOT have any existing nodes with same ID as nodeA. 
```
My question is: If a new node is created on canvas B at a later point in time, will SVG.js prevent any duplicate ID?

**Comment by Fuzzyma** [comment]
There is no mechanism which checks if an is is already used on the site. So it wont save you from doing stuff like that. However, when you clone a node a new id is assigned automatically

**Comment by Zythyr** [comment]
Thanks. Issue resolved.
